### PR TITLE
Refactor Application Gateway to expose authorisation related code 

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- Authorisation and CSRF related code moved to CSRF folder
- Code for setting additional headers/query parameters moved to pkg folder 

**Related issue(s)**

### CODE REVIEW

File: components/application-gateway/pkg/authorization/model.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	// BasicAuth is BasicAuth configuration.
+	BasicAuth *BasicAuth
+	// CertificateGen is CertificateGen configuration.
+	CertificateGen *CertificateGen
+	// CSRFTokenEndpointURL (optional) to fetch CSRF token
+	CSRFTokenEndpointURL string
+	// Headers that are injected by the gateway
+	Headers *map[string][]string
+	// QueryParameters that are injected by the gateway
+	QueryParameters *map[string][]string
```
Comment (Szymongib) [inline components/application-gateway/pkg/authorization/model.go]: As discussed `Headers` and `QueryParameters` are leftovers and we should remove those.

Comment (akgalwas) [reply components/application-gateway/pkg/authorization/model.go]: Done

**Pull request merged**